### ISSUE TITLE
The reporter option causes tests to always pass

### ISSUE THREAD

**Issue by msteen** [dubzzz/jest-fast-check#122]
I am using my own reporter function to have more details about what goes wrong with my tests and prettyprint the counterexamples and such, but unfortunately it seems that this always causes the test to pass. Initially I was just using `fc.assert` with Jest, and this problem caused me to use this project, but unfortunately it still seems present. Here is an example that should always fail, yet passes:

```typescript
import * as fc from "fast-check"
import { testProp } from "jest-fast-check"

testProp(
  "always fails",
  [fc.boolean()],
  (_boolean) => {
    return false
  },
  {
    reporter() {
      console.log("report")
    },
  },
)
```

These are my dev dependencies:
```json
{
  "@types/jest": "^27.0.2",
  "@types/node": "^16.11.6",
  "fast-check": "^2.19.0",
  "jest": "^27.3.1",
  "jest-fast-check": "^1.0.2",
  "prettier": "^2.4.1",
  "ts-jest": "^27.0.7",
  "ts-node": "^10.4.0",
  "typescript": "^4.4.4"
}
```

I could of course write my own wrapper to make this work, but AFAIK that is this project's purpose already, so I thought it best to report it here.

**Comment by dubzzz** [comment]
Hey,

In theory, the reporter has to handle the failure case on its own. In other words, it should throw in case there is something that should be considered as a failure. There is not so much documentation on this attribute but here is the one I found: https://github.com/dubzzz/fast-check/blob/main/documentation/Tips.md#customize-the-reported-error

You probably want to start from a reporter like this one and enhance it with your custom behaviours:
```
  {
    reporter(out) {
      // Let's say we want to re-create the default reporter of `assert`
      if (out.failed) {
        // `defaultReportMessage` is an utility that make you able to have the exact
        // same report as the one that would have been generated by `assert`
        throw new Error(fc.defaultReportMessage(out));
      }
    }
  }
```

If I'm not wrong I did not want to handle the throw on custom reporters as it may forbid advanced users to use it on their needs and force them to ask yet for another settings to do so. As a consequence the `reporter` is fully manual.

Regards,
Nicolas

**Comment by msteen** [comment]
Thanks for the clarification! I should have better read the documentation. I think I just found out about reporter using autocomplete and then it worked based on my assumptions (I just reported what went wrong via logging), which worked fine outside a test suite, but with e.g. Jest, those assumptions broke. In hindsight, I even see it mentioned when I further inspect the in-editor documentation about it:
> Reporter is responsible to throw in case of failure: default one throws whenever runDetails.failed is true. But you may want to change this behaviour in yours.
